packages/react/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,426 @@
+  }
+
+  itemRefs = [] as React.RefObject<HTMLElement>[]
+  paddleNextRef = React.createRef<HTMLElement>()
+  paddlePreviousRef = React.createRef<HTMLElement>()
+
+  focusItemAtIndex = debounce((index: number) => {
+    this.itemRefs[index].current.focus()
+  }, 400)
+
+  setActiveIndex(index: number, focusItem: boolean): void {
+    const { circular, items } = this.props
+    const lastItemIndex = items.length - 1
+    let activeIndex = index
+
+    if (index < 0) {
+      if (!circular) {
+        return
+      }
+      activeIndex = lastItemIndex
+    }
+
+    if (index > lastItemIndex) {
+      if (!circular) {
+        return
+      }
+      activeIndex = 0
+    }
+
+    this.setState({
+      activeIndex,
+    })
+
+    if (focusItem) {
+      this.focusItemAtIndex(activeIndex)
+    }
+  }
+
+  renderContent = (accessibility, styles, unhandledProps) => {
+    const { ariaRoleDescription, getItemPositionText, items } = this.props
+    const { activeIndex, itemIds } = this.state
+
+    this.itemRefs = []
+
+    return (
+      <div style={styles.itemsContainerWrapper} {...accessibility.attributes.itemsContainerWrapper}>
+        <div
+          className={Carousel.slotClassNames.itemsContainer}
+          aria-roledescription={ariaRoleDescription}
+          style={styles.itemsContainer}
+          {...accessibility.attributes.itemsContainer}
+          {...applyAccessibilityKeyHandlers(
+            accessibility.keyHandlers.itemsContainer,
+            unhandledProps,
+          )}
+        >
+          {items &&
+            items.map((item, index) => {
+              const itemRef = React.createRef<HTMLElement>()
+              this.itemRefs.push(itemRef)
+              return (
+                <Ref key={item['key'] || index} innerRef={itemRef}>
+                  {CarouselItem.create(item, {
+                    defaultProps: () => ({
+                      active: activeIndex === index,
+                      id: itemIds[index],
+                      ...(getItemPositionText && {
+                        itemPositionText: getItemPositionText(index, items.length),
+                      }),
+                    }),
+                  })}
+                </Ref>
+              )
+            })}
+        </div>
+      </div>
+    )
+  }
+
+  showPreviousSlide = (e: React.SyntheticEvent, focusItem: boolean) => {
+    this.setActiveIndex(this.state.activeIndex - 1, focusItem)
+  }
+
+  showNextSlide = (e: React.SyntheticEvent, focusItem: boolean) => {
+    this.setActiveIndex(this.state.activeIndex + 1, focusItem)
+  }
+
+  renderPaddles = (accessibility, styles) => {
+    const { paddlePrevious, paddleNext } = this.props
+
+    return (
+      <>
+        <Ref innerRef={this.paddlePreviousRef}>
+          {Button.create(paddlePrevious, {
+            defaultProps: () => ({
+              className: Carousel.slotClassNames.paddlePrevious,
+              iconOnly: true,
+              icon: 'stardust-chevron-left',
```

Comment:
let's add these in the defaultProps, instead of the empty objects.

Reply:
If I do this, it will override all of them each time I add only one field. For instance in the pagination example I added aria-label as shortcuts and the default values were removed. I would consider leaving theme here.